client/components/tinymce/plugins/simple-payments/dialog/form.jsx

```diff
@@ -42,6 +51,12 @@ const validate = ( values, props ) => {
 
 	if ( ! values.price ) {
 		errors.price = translate( 'Price can not be empty.' );
+	} else if ( parseFloat( values.price ) === NaN ) {
+		errors.price = translate( 'Price is malformed.' );
+	} else if ( parseFloat( values.price ) < 0 ) {
+		errors.price = translate( 'Price can not be negative.' );
+	} else if ( decimalPlaces( values.price ) > 2 ) {
+		errors.price = translate( 'Price can not have higher than 2 decimal places.' );
```

Comment:
The number of decimal places actually depends on the currency. Some currencies have zero, some have three. See `client/lib/format-currency/currencies.js`.

The form component already gets a `currencyDefaults` prop and the precision is in the `currencyDefaults.precision` field.

Instead of "higher than", I'd say "more than", but I might be wrong :)